source/hp/dof_handler.cc

```diff
@@ -1868,19 +1972,113 @@ namespace hp
               }
           }
 
+        // Clear stored active_fe_indices.
+        refined_cells_fe_index.clear();
+        coarsened_cells_fe_index.clear();
+      }
+  }
+
+
+  template <int dim, int spacedim>
+  void
+  DoFHandler<dim, spacedim>::post_shared_refinement_fe_index_transfer()
+  {
+    // Finite elements need to be assigned to each cell by calling
+    // distribute_dofs() first to make this functionality available.
+    if (fe_collection.size() > 0)
```

Comment:
oh, I see the answer to my questions earlier on. This is awkward. You might want to explain in the code snippet I annotated why it is necessary to distribute dof indices...

Reply:
I guess what we really need is a `set_fe()` function that just sets the FE collection but doesn't actually distribute DoFs -- but that's a substantially larger change to ensure that things continue to be internally consistent. So maybe don't do this right now...

Reply:
A function like that would be handy for this case. We need to attach the `FECollection` before assigning `active_fe_indices` in any case.

I'll open an issue for this.